src/System.IO.Pipelines/ReadableBufferReader.cs

```diff
@@ -80,7 +86,7 @@ public int Take()
 
         private void MoveNext()
         {
-            if (_enumerator.MoveNext())
+            if (_enumerator.MoveNext() && !_enumerator.Current.IsEmpty)
```

Comment:
How is this correct? Having empty segment in the middle of the pipe should work fine.